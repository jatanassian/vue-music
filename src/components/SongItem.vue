<template>
  <div class="border border-gray-200 p-3 mb-4 rounded">
    <div v-show="!showEditForm">
      <h4 class="inline-block text-2xl font-bold">{{ song.modified_name }}</h4>
      <button class="ml-1 py-1 px-2 text-sm rounded text-white bg-red-600 float-right">
        <i class="fa fa-times"></i>
      </button>
      <button
        class="ml-1 py-1 px-2 text-sm rounded text-white bg-blue-600 float-right"
        @click="showEditForm = true"
      >
        <i class="fa fa-pencil-alt"></i>
      </button>
    </div>
    <div v-show="showEditForm">
      <VeeForm :initial-values="song" @submit="editSong">
        <div class="mb-3">
          <label class="inline-block mb-2">Song Title</label>
          <VeeField
            name="modified_name"
            type="text"
            class="block w-full py-1.5 px-3 text-gray-800 border border-gray-300 transition duration-500 focus:outline-none focus:border-black rounded"
            placeholder="Enter Song Title"
            rules="required"
          />
          <ErrorMessage class="text-red-600" name="modified_name" />
        </div>
        <div class="mb-3">
          <label class="inline-block mb-2">Genre</label>
          <VeeField
            name="genre"
            type="text"
            class="block w-full py-1.5 px-3 text-gray-800 border border-gray-300 transition duration-500 focus:outline-none focus:border-black rounded"
            placeholder="Enter Genre"
          />
        </div>
        <button type="submit" class="py-1.5 px-3 rounded text-white bg-green-600">Submit</button>
        <button
          type="button"
          class="py-1.5 px-3 rounded text-white bg-gray-600"
          @click="showEditForm = false"
        >
          Go Back
        </button>
      </VeeForm>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SongItem',
  props: {
    song: { type: Object, required: true }
  },
  data() {
    return {
      showEditForm: false
    };
  },
  methods: {
    editSong(formValues) {
      console.log(formValues);
    }
  }
};
</script>
