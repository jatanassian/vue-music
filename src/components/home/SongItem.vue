<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
    :data-testid="`song-id-${song.id}`"
  >
    <div>
      <RouterLink
        :to="{ name: 'song', params: { id: song.id } }"
        class="font-bold block text-gray-600"
        data-cy="song-name"
      >
        {{ song.modified_name }}
      </RouterLink>
      <span class="text-gray-500 text-sm">{{ song.display_name }}</span>
    </div>

    <div class="text-gray-600 text-lg">
      <RouterLink
        custom
        :to="{ name: 'song', params: { id: song.id }, hash: '#comments' }"
        v-slot="{ navigate }"
      >
        <span class="comments" @click="navigate">
          <i class="fa fa-comments text-gray-600"></i>
          {{ song.comments_count }}
        </span>
      </RouterLink>
    </div>
  </li>
</template>

<script>
export default {
  name: 'SongItem',
  props: { song: { type: Object, required: true } }
};
</script>
