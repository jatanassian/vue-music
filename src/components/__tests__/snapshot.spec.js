import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import SongItem from '../home/SongItem.vue';

describe('Snapshots SongItem.vue', () => {
  test('Renders correctly', () => {
    const song = {
      id: 'abc',
      display_name: 'Test',
      modifier_name: 'Test',
      comments_count: 5
    };

    const component = shallowMount(SongItem, {
      props: { song },
      global: {
        components: { RouterLink: RouterLinkStub }
      }
    });

    expect(component.element).toMatchSnapshot();
  });
});
