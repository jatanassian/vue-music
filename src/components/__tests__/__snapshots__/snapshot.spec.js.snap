// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshots SongItem.vue > Renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  data-testid="song-id-abc"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      Test
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
